backend/substrapp/utils.py

```diff
@@ -108,6 +108,25 @@ def create_directory(directory):
         os.makedirs(directory)
 
 
+def raise_if_path_traversal(requested_paths, to_directory):
+    # Inspired from https://stackoverflow.com/a/45188896
+
+    safe_directory = os.path.join(
+        os.path.realpath(to_directory),
+        ''
+    )
+
+    if not isinstance(requested_paths, list):
+        raise TypeError(f'requested_paths argument should be a list not a {type(requested_paths)}')
+
+    for requested_path in requested_paths:
+        real_requested_path = os.path.realpath(requested_path)
+        path_traversal = os.path.commonprefix([real_requested_path, safe_directory]) != safe_directory
```

Comment:
Same comment as in https://github.com/SubstraFoundation/substra/pull/151 i'm not sure that this code really prevent from path traversal.